pkg/apis/resource/v1alpha1/git/git_resource_test.go

```diff
@@ -675,6 +675,44 @@ func TestGitResource_GetDownloadTaskModifier(t *testing.T) {
 				{Name: "NO_PROXY", Value: "no-proxy.git.com"},
 			},
 		},
+	}, {
+		desc: "Without Refspec and without revision",
+		gitResource: &git.Resource{
+			Name:       "git-resource",
+			Type:       resourcev1alpha1.PipelineResourceTypeGit,
+			URL:        "git@github.com:test/test.git",
+			Revision:   "",
+			Refspec:    "",
+			GitImage:   "override-with-git:latest",
+			Submodules: false,
+			Depth:      1,
+			SSLVerify:  true,
+			HTTPProxy:  "http-proxy.git.com",
+			HTTPSProxy: "https-proxy.git.com",
+			NOProxy:    "no-proxy.git.com",
+		},
+		want: corev1.Container{
+			Name:    "git-source-git-resource-twkr2",
+			Image:   "override-with-git:latest",
+			Command: []string{"/ko-app/git-init"},
+			Args: []string{
+				"-url",
+				"git@github.com:test/test.git",
+				"-revision",
```

Comment:
nit: should we just leave the flag out completely if there's no revision?

Reply:
I tried removing this but the unit test is failing :sweat_smile:

Reply:
Indeed, you would also need to update line 149 in `pkg/apis/resource/v1alpha1/git/git_resource.go` to only add the `-revision` flag if `s.Revision != ""`

Reply:
thanks :sweat_smile: